web/src/components/text-input/styles.ts

```diff
@@ -0,0 +1,85 @@
+import { createUseStyles } from 'react-jss'
+
+import tokens from '../../tokens'
+import { getColorDefault, getColorHover } from '../checkbox/styles'
+
+export default createUseStyles({
+  textField: {
+    fontFamily: tokens.fontFamily,
+    borderRadius: ({ component }) => {
+      return component === 'input'
+        ? tokens.borderRadiusFormfieldInput
+        : tokens.borderRadiusFormfieldTextarea
+    },
+    borderWidth: tokens.borderWidthFormfield,
+    borderStyle: 'solid',
+    color: tokens.colorTextPrimary,
+    resize: 'none',
+    backgroundColor: ({ isDisabled, isReadOnly }) => {
+      if (isDisabled) return tokens.colorBackgroundSecondary
+      if (isReadOnly) return tokens.colorBackgroundFormfieldReadonly
+      return undefined
```

Comment:
Maybe better to return `white` here?

Reply:
Should we return the token for the default formfield background?

Reply:
Also for disabled we should use the specified token. I just noticed a change in the design tokens to be made (change disabled-hgost from "gray-5" to "gray-2"). I'll do this then to look won't change and you can use the token.